### PR TITLE
Add TS types definitions

### PR DESCRIPTION
**Pull request by krzysztofciombor**
## Description
Adds TypeScript types definitions.
See also: #78 

## Test plan
The TS types are correctly picked up when importing Gesture Handlers inside `*.ts` file:
zrzut ekranu 2017-12-19 o 17 17 25
zrzut ekranu 2017-12-19 o 17 17 34

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-  "description":
-    "Experimental implementation of a new declarative API for gesture handling in react-native",
+  "description": "Experimental implementation of a new declarative API for gesture handling in react-native",
   "scripts": {
     "start": "node node_modules/react-native/local-cli/cli.js start",
     "test": "jest",
     "precommit": "lint-staged"
   },
   "main": "GestureHandler.js",
+  "types": "react-native-gesture-handler.d.ts",
```
Comment (kmagiera) [inline package.json]: I think this needs to be added to "files" section too

**Comment by brunolemos** [comment]
It works if I import like this: `import { Swipeable } from 'react-native-gesture-handler'`
But not if I import like this: `import Swipeable from 'react-native-gesture-handler/Swipeable'`

**Comment by brunolemos** [comment]
#87 should fix it

**Comment by brunolemos** [comment]
I get this on `RectButton` and don't know why:

```
[ts] Property 'style' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<RectButton> & Readonly<{ children?: ReactNode; }> ...'.
(JSX attribute) style: ((number & {
    __registeredStyleBrand: ViewStyle | TextStyle | ImageStyle;
}) | {
    backgroundColor: string;
    width: number | undefined;
})[]
```

image

**Pull request merged**